src/MQTTAsync.c

```diff
@@ -329,6 +329,9 @@ int MQTTAsync_createWithOptions(MQTTAsync* handle, const char* serverURI, const
 	{
 		if (strncmp(URI_TCP, serverURI, strlen(URI_TCP)) != 0
 		 && strncmp(URI_MQTT, serverURI, strlen(URI_MQTT)) != 0
+#if defined(UNIXSOCK)
```

Comment:
you should avoid such `if defined()` in order to avoid code complexity. The error cases can be handled for WIN32 or non UNIXSOCK into the function `Socket_unix_new()` that can be available for all the cases, but that should return a runtime ERROR when Unixsock is not available. 

So the code will be easier to read.

Reply:
I disagree here. The conditional compilation ensures that `unix://` is reported as an unsupported protocol if support is not compiled into the library - just the same way it's done with  `ssl://`, `mqtts://`, and `wss://`.